ui/ducks/send/send.js

```diff
@@ -1512,26 +1514,28 @@ const slice = createSlice({
         state.selectedAccount.balance = action.payload.account.balance;
         const draftTransaction =
           state.draftTransactions[state.currentTransactionUUID];
-        draftTransaction.gas.gasLimit = action.payload.gasLimit;
+        if (draftTransaction) {
+          draftTransaction.gas.gasLimit = action.payload.gasLimit;
+          draftTransaction.gas.gasTotal = action.payload.gasTotal;
+          if (action.payload.chainHasChanged) {
+            // If the state was reinitialized as a result of the user changing
+            // the network from the network dropdown, then the selected asset is
+            // no longer valid and should be set to the native asset for the
+            // network.
+            draftTransaction.asset.type = ASSET_TYPES.NATIVE;
+            draftTransaction.asset.balance =
+              draftTransaction.fromAccount?.balance ??
+              state.selectedAccount.balance;
+            draftTransaction.asset.details = null;
+          }
+        }
```

Comment:
So the only way this code should be hit is if there was an inflight attempt to initialize the send state, which at the time had a valid draftTransaction, and then the send flow is reset *between when the initialize is fulfilled and when redux runs this case* the only way this could be done (to my knowledge) is by changing networks in the send flow and canceling and hope you get lucky to time it right?